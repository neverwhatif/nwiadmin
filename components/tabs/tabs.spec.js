import React from 'react';
import { shallow } from 'enzyme';

import { TabsComponent } from './index';

jest.mock('./tabitem', () => 'TabItem');

const mockLocation = {
    pathname: '/lorem',
    search: '',
};

const mockUpdatedLocation = {
    pathname: '/ipsum',
    search: '',
};

const data = [{
    key: 0,
    label: 'Lorem',
    path: 'lorem',
}, {
    key: 1,
    label: 'Ipsum',
    path: 'ipsum',
}];

describe('TabsComponent', () => {
    const component = shallow(<TabsComponent data={data} basePath="lorem" location={mockLocation} />);

    it('should render', () => {
        expect(component).toMatchSnapshot();
    });
    it('should render tab items', () => {
        expect(component.find('li').length).toBe(2);
        expect(component.find('TabItem').length).toBe(2);
    });
    it('should run shouldComponentUpdate', () => {
        const shouldNotUpdate = component.instance().shouldComponentUpdate({ location: mockLocation });
        expect(shouldNotUpdate).toBe(false);

        const shouldUpdate = component.instance().shouldComponentUpdate({ location: mockUpdatedLocation });
        expect(shouldUpdate).toBe(true);
    });
});
