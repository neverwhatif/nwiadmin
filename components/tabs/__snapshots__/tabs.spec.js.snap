// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TabsComponent should render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <TabsComponent
    basePath="lorem"
    data={
      Array [
        Object {
          "key": 0,
          "label": "Lorem",
          "path": "lorem",
        },
        Object {
          "key": 1,
          "label": "Ipsum",
          "path": "ipsum",
        },
      ]
    }
    location={
      Object {
        "pathname": "/lorem",
        "search": "",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <li
          className="item"
        >
          <TabItem
            basePath="lorem"
            isActive={false}
            label="Lorem"
            path="/lorem/lorem"
          />
        </li>,
        <li
          className="item"
        >
          <TabItem
            basePath="lorem"
            isActive={false}
            label="Ipsum"
            path="/lorem/ipsum"
          />
        </li>,
      ],
      "className": "root",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": "0",
        "nodeType": "host",
        "props": Object {
          "children": <TabItem
            basePath="lorem"
            isActive={false}
            label="Lorem"
            path="/lorem/lorem"
          />,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "basePath": "lorem",
            "isActive": false,
            "label": "Lorem",
            "path": "/lorem/lorem",
          },
          "ref": null,
          "rendered": null,
          "type": "TabItem",
        },
        "type": "li",
      },
      Object {
        "instance": null,
        "key": "1",
        "nodeType": "host",
        "props": Object {
          "children": <TabItem
            basePath="lorem"
            isActive={false}
            label="Ipsum"
            path="/lorem/ipsum"
          />,
          "className": "item",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "basePath": "lorem",
            "isActive": false,
            "label": "Ipsum",
            "path": "/lorem/ipsum",
          },
          "ref": null,
          "rendered": null,
          "type": "TabItem",
        },
        "type": "li",
      },
    ],
    "type": "ul",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <li
            className="item"
          >
            <TabItem
              basePath="lorem"
              isActive={false}
              label="Lorem"
              path="/lorem/lorem"
            />
          </li>,
          <li
            className="item"
          >
            <TabItem
              basePath="lorem"
              isActive={false}
              label="Ipsum"
              path="/lorem/ipsum"
            />
          </li>,
        ],
        "className": "root",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": "0",
          "nodeType": "host",
          "props": Object {
            "children": <TabItem
              basePath="lorem"
              isActive={false}
              label="Lorem"
              path="/lorem/lorem"
            />,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "0",
            "nodeType": "host",
            "props": Object {
              "basePath": "lorem",
              "isActive": false,
              "label": "Lorem",
              "path": "/lorem/lorem",
            },
            "ref": null,
            "rendered": null,
            "type": "TabItem",
          },
          "type": "li",
        },
        Object {
          "instance": null,
          "key": "1",
          "nodeType": "host",
          "props": Object {
            "children": <TabItem
              basePath="lorem"
              isActive={false}
              label="Ipsum"
              path="/lorem/ipsum"
            />,
            "className": "item",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": "1",
            "nodeType": "host",
            "props": Object {
              "basePath": "lorem",
              "isActive": false,
              "label": "Ipsum",
              "path": "/lorem/ipsum",
            },
            "ref": null,
            "rendered": null,
            "type": "TabItem",
          },
          "type": "li",
        },
      ],
      "type": "ul",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
